src/libraries/Microsoft.Extensions.DependencyInjection/src/DependencyInjectionEventSource.cs

```diff
@@ -27,9 +27,11 @@ public static class Keywords
 
         private readonly List<WeakReference<ServiceProvider>> _providers = new();
 
+#pragma warning disable ESGEN001 // EventSource classes should not declare constructors.
```

Comment:
Is there a tracking issue for these?

Reply:
Added a link to https://github.com/dotnet/runtime/pull/121180 which is effectively the reason